### PR TITLE
ENG-1101 User Be Able To Download Documents After Mint

### PR DESCRIPTION
**Pull request by angelvazz**
- Add keys to every map.
- Add a function to download every file after minting.
- Add a map function to generate a card with a download button to download the specific file.

### CODE REVIEW

File: packages/apps/MintingB2B/src/pages/NFTDetails/index.tsx [hunk 1]
```diff
@@ -139,6 +141,28 @@ const WalletPage = () => {
...
+    });
+    const data = await response.blob();
+
+    const downloadUrl = URL.createObjectURL(data);
+    const a = document.createElement('a');
+    a.href = downloadUrl;
+    a.download = fileName;
+    document.body.appendChild(a);
+    a.click();
+    document.body.removeChild(a);
```
Comment (Motokoder) [inline packages/apps/MintingB2B/src/pages/NFTDetails/index.tsx:163]: Can you check if the link flashes on and off the screen with different browsers (chrome, safari, firefox)?
If so, maybe you can position it off-screen. I don't think you can click a link with JavaScript if display is set to none or visibility false.

**Review by Motokoder — CHANGES_REQUESTED** [review]

File: packages/apps/MintingB2B/src/pages/NFTDetails/index.tsx [hunk 2]
```diff
@@ -184,46 +209,39 @@ const WalletPage = () => {
...
+                        )}
+                        {normalData.status === 'WAITING_FOR_OWNER' && (
+                          <Grid columns={2} gap={16}>
+                            <Button onClick={generateQR} btnType="filled" style={{ width: '100%' }}>
+                              Generate QR
+                            </Button>
+                            <Button
+                              onClick={() => setIsOpen(true)}
+                              btnType="outlined"
+                              disabled={normalData.status !== 'WAITING_FOR_OWNER'}
```
Comment (Motokoder) [inline packages/apps/MintingB2B/src/pages/NFTDetails/index.tsx:230]: Please replace any repeating status strings with a constant?

File: packages/apps/MintingB2B/src/pages/NFTDetails/index.tsx (hunk 1 above)
Comment (angelvazz) [reply packages/apps/MintingB2B/src/pages/NFTDetails/index.tsx:165]: Test on Chrome, Safari and FireFox: Nothing behaves different.

File: packages/apps/MintingB2B/src/pages/NFTDetails/index.tsx (hunk 2 above)
Comment (angelvazz) [reply packages/apps/MintingB2B/src/pages/NFTDetails/index.tsx]: Strings changed to const and created the switch statement for the NFT Status

**Review by Motokoder — CHANGES_REQUESTED** [review]

File: packages/apps/MintingB2B/src/pages/NFTDetails/index.tsx [hunk 3]
```diff
@@ -63,8 +65,10 @@ const WalletPage = () => {
   const [isSendOpen, setIsSendOpen] = useState(false);
   const { nft_id } = useParams();
   const [normalData, setNormalData] = useState<any>();
+  const { waitingForOwner, success, preStage } = STATUSES;
```
[the remark this answers is not in the document]
Comment (angelvazz) [reply packages/apps/MintingB2B/src/pages/NFTDetails/index.tsx]: All conditional statements fixed.

**Pull request merged**